src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const stateResult = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        newState = { ...newState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        newState = { ...newState };
```

Comment:
[CODE STYLE]: There is no need to make a copy of `newState` with the spread operator here. You are already working with a copy of the state, so you can just remove properties directly from it.